### PR TITLE
Drop support for Python 3.4 (#86)

### PR DESCRIPTION
**Pull request by csadorf**
Drop support for Python version 3.4.

Resolves issue #86 .

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,4 @@
 signac>=0.9.0
-enum34 ; python_version < "3.4"
```
Comment (bdice) [inline requirements.txt]: This line is actually for Python 2.7, not 3.4 if I understand correctly. https://github.com/conda-forge/signac-flow-feedstock/blob/a99c7d3728dbae4d8534c2797d37b64fb054c509/recipe/meta.yaml#L24

File: setup.py [hunk 2]
```diff
@@ -12,8 +12,6 @@
     'jinja2>=2.8',
     # To enable the parallelized execution of operations across processes.
     'cloudpickle',
-    # To define IntEnum in flow/scheduling/base.py.
-    'enum34;python_version<"3.4"',
```
Comment (bdice) [inline setup.py]: Same as above, may actually be for 2.7.

**Comment by bdice** [comment]
@csadorf I reviewed too fast 😆

**Comment by csadorf** [comment]
> @csadorf I reviewed too fast 😆

I was too slow. ;)

**Pull request merged**